src/components/line-chart-audification/line-chart-audification.component.ts

```diff
@@ -1,25 +1,116 @@
-import { Component, Host } from '@angular/core';
-import { LineChartComponent } from '../line-chart/line-chart.component';
-import { AudificationService } from '../../services/audification/audification.service';
-import * as Tone from 'tone';
+import { Component, EventEmitter, HostListener, Input, NgZone, OnChanges, OnDestroy, Output, SimpleChanges } from '@angular/core';
+import { Melody } from '../../models/melody/melody.model';
+import { AUDIFICATION, t, tA11y } from '../../assets/i18n';
+import { Datum } from '../../d3/xy-chart.d3';
+import { formatX, formatY } from '../../utils/formatters';
 
 @Component({
   selector: 'app-line-chart-audification',
   templateUrl: './line-chart-audification.component.html',
+  styleUrls: ['./line-chart-audification.component.scss'],
 })
-export class LineChartAudificationComponent {
-  public disposeMelody?: () => void;
+export class LineChartAudificationComponent implements OnDestroy, OnChanges {
+  @Input() data: Datum[];
+  @Input() activeDatum: Datum | null;
+  @Output() activeDatumChange = new EventEmitter<Datum | null>();
+  @Input() frequencyRange: [number, number] = [256, 2048];
+  @Input() duration = 5;
+  liveText: string | null = null;
+  private melody?: Melody;
+  private domain: Date[];
+  private range: number[];
 
   constructor(
-    @Host() public component: LineChartComponent,
-    private audificationService: AudificationService,
+    private zone: NgZone,
   ) {
+    this.handleSeek = this.handleSeek.bind(this);
   }
 
-  async playMelody() {
-    if (Tone.getContext().state === 'suspended') {
-      await Tone.start();
+  get INSTRUCTIONS() {
```

Comment:
I'm a bit torn on using the custom i18n solution inside an Angular component, since Angular has their own i18n features.
https://angular.io/guide/i18n

I really like the i18n custom attribute and find it makes templates super easy to understand. It would be nice to have the text displayed in the HTML and not have to track down AUDIFICATION.INSTRUCTIONS to see what it actually says.

With that said, using two different systems of internationalization increases the translation complexity, so it makes sense to continue to use the custom i18n translate method used in the D3 files for this project.

Basically - I guess this wasn't a comment for change so much as me just musing aloud. :)

Reply:
I see the pros and cons of both methods. Do you know how GA is handling i18n in d3 manipulation?